runtime/src/bank/partitioned_epoch_rewards/mod.rs

```diff
@@ -39,14 +44,98 @@ pub(crate) struct PartitionedStakeReward {
     pub commission: u8,
 }
 
-type PartitionedStakeRewards = Vec<PartitionedStakeReward>;
+/// A vector of stake rewards.
+#[derive(Debug, Default)]
+pub(crate) struct PartitionedStakeRewards {
+    /// Inner vector.
+    rewards: Vec<Option<PartitionedStakeReward>>,
+    /// Number of `Some` elements.
+    len_some: usize,
+}
+
+impl PartitionedStakeRewards {
+    pub(crate) fn with_capacity(capacity: usize) -> Self {
+        let rewards = Vec::with_capacity(capacity);
+        Self {
+            rewards,
+            len_some: 0,
+        }
+    }
+
+    pub(crate) fn capacity(&self) -> usize {
+        self.rewards.capacity()
+    }
+
+    /// Number of `Some` elements.
+    pub(crate) fn len_some(&self) -> usize {
+        self.len_some
+    }
+
+    pub(crate) fn get(&self, index: usize) -> Option<&Option<PartitionedStakeReward>> {
+        self.rewards.get(index)
+    }
+
+    pub(crate) fn iter(&self) -> Iter<'_, Option<PartitionedStakeReward>> {
+        self.into_iter()
+    }
+
+    fn spare_capacity_mut(&mut self) -> &mut [MaybeUninit<Option<PartitionedStakeReward>>] {
+        self.rewards.spare_capacity_mut()
+    }
+
+    unsafe fn set_len(&mut self, new_len: usize) {
+        self.rewards.set_len(new_len)
+    }
+
+    fn set_len_some(&mut self, len_some: usize) {
+        self.len_some = len_some
+    }
+}
+
+impl FromIterator<Option<PartitionedStakeReward>> for PartitionedStakeRewards {
+    fn from_iter<T: IntoIterator<Item = Option<PartitionedStakeReward>>>(iter: T) -> Self {
+        let mut len_some: usize = 0;
+        let rewards = Vec::from_iter(iter.into_iter().inspect(|reward| {
+            if reward.is_some() {
+                len_some = len_some.saturating_add(1);
+            }
+        }));
+        Self { rewards, len_some }
+    }
+}
+
+impl<I> Index<I> for PartitionedStakeRewards
+where
+    I: SliceIndex<[Option<PartitionedStakeReward>]>,
+{
+    type Output = I::Output;
+
+    fn index(&self, index: I) -> &Self::Output {
+        self.rewards.index(index)
+    }
+}
+
+impl<'a> IntoIterator for &'a PartitionedStakeRewards {
+    type Item = <Self::IntoIter as IntoIterator>::Item;
+    type IntoIter = Iter<'a, Option<PartitionedStakeReward>>;
+
+    fn into_iter(self) -> Self::IntoIter {
+        self.rewards.iter()
+    }
+}
+
+impl PartialEq for PartitionedStakeRewards {
+    fn eq(&self, other: &Self) -> bool {
+        self.rewards.eq(&other.rewards)
+    }
+}
```

Comment:
I think it's fine to derive this trait and a little more correct to include the `len_some` field anyways. What do you think?

Reply:
Yes, I agree, I derived it now